cmd/gpu_plugin/gpu_plugin_test.go

```diff
@@ -696,3 +714,253 @@ func TestBypath(t *testing.T) {
 		}
 	}
 }
+
+func TestPciDeviceForCard(t *testing.T) {
+	root, err := os.MkdirTemp("", "test_pci_device_for_card")
+	if err != nil {
+		t.Fatalf("can't create temporary directory: %+v", err)
+	}
+	// dirs/files need to be removed for the next test
+	defer os.RemoveAll(root)
+
+	sysfs := path.Join(root, "sys")
+
+	cardPath := filepath.Join(sysfs, "class", "drm", "card0")
+	cardDevicePath := filepath.Join(cardPath, "device")
+
+	if err := os.MkdirAll(cardDevicePath, 0750); err != nil {
+		t.Fatalf("couldn't create card's device path")
+	}
+
+	data := "0x5959"
+
+	err = os.WriteFile(filepath.Join(cardDevicePath, "device"), []byte(data), 0o600)
+	if err != nil {
+		t.Fatalf("device id write failed: %+v", err)
+	}
+
+	id, err := pciDeviceIDForCard(cardPath)
+
+	if err != nil {
+		t.Error("failed to get device id for card")
+	}
+
+	if id != data {
+		t.Errorf("wrong id received %s vs %s", id, data)
+	}
+
+	// Check bad device
+
+	cardPath = filepath.Join(sysfs, "class", "drm", "card1")
+	cardDevicePath = filepath.Join(cardPath, "device")
+
+	if err := os.MkdirAll(cardDevicePath, 0750); err != nil {
+		t.Fatalf("couldn't create card's device path")
+	}
+
+	err = os.WriteFile(filepath.Join(cardDevicePath, "devicee"), []byte(data), 0o600)
```

Comment:
"devicee"?
```suggestion
	err = os.WriteFile(filepath.Join(cardDevicePath, "device"), []byte(data), 0o600)
```

Reply:
This is a negative test so it's on purpose. But I'll rename the file so it's more obvious.